test/util/log/log.go

```diff
@@ -0,0 +1,89 @@
+package log
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	logrus_test "github.com/sirupsen/logrus/hooks/test"
+)
+
+// ExpectedLogEntry contains a log message and log level which is expected to be
+// emitted by the logging system.
+type ExpectedLogEntry struct {
+	// The message to be matched exactly. Conflicts with MessageRegex.
+	Message string
+
+	// The message to be matched as regex. Conflicts with Message.
+	MessageRegex string
+
+	// The logging level to be matched.
+	Level logrus.Level
+}
+
+// assertMatches compares the expected entry with an actual Entry. An error is
+// returned if they do not match.
+func (ex ExpectedLogEntry) assertMatches(e logrus.Entry) string {
```

Comment:
The comment says an error is returned, but actually a string is. I'd suggest either returning an error or changing the comment to say "Error text", although the latter seems strange to me.

Reply:
Yeah. Cleaned this up. (It's also in https://github.com/Azure/ARO-RP/pull/870 with the goal of shrinking this PR)